### ISSUE TITLE
Accessing SQLite databases within ZIP files

### ISSUE THREAD

**Issue by ww-9** [sqlitebrowser/sqlitebrowser#2369]
Describe the new feature
--------------------------
Compression of database files allows to achieve a significant space savings. It would be extremely convenient to immediately have access to such zipped database files without the need to unzip first.

Additional info
---------------
#### Does this feature exist in another product or project? Please provide a link.
https://sqlite.org/zipfile.html

**Comment by justinclift** [comment]
Interesting idea.  Reading through the zipfile page you linked, it seems more link a SQLite extension that reads in .zip files directly then presents the file data as SQLite tables.

eg not a "read in compressed SQLite file" thing.

To me, both concepts have merit.  eg:

1. Add support for reading compressed (zip, bz2, xz, etc) sqlite databases, automatically re-compressing when saving them back out.  Personally I'd start with zip or gz, as the "save" part is still pretty fast.  Higher compression things like bz2, and xz are very slow for compressing.

2. We can try including the zipfile extension.  No idea how that'll go, but we already include other SQLite extensions for it might not be too hard. :smile:

**Comment by ww-9** [comment]
Yes, that zipfile extension has slightly different functionality.

The zip was proposed because it has indexes so the only required db can be accessed without having to unpack the entire archive. However, with modern amounts of memory this is not a huge problem.
